spec/1.1/index.bs

```diff
@@ -210,7 +210,7 @@ Functionally equivalent to <a href="https://www.w3.org/TR/html51/webappapis.html
 Passing the value returned by {{requestAnimationFrame()}} to will unregister the callback.
 
 <dfn method for="VRDisplay">requestPresent()</dfn>
-Begins presenting the contents of the specified {{VRLayerInit}} array on the {{VRDisplay}} and fulfills the returned promise when presentation has begun. If {{canPresent}} is false the promise MUST be rejected. If the {{VRLayerInit}} array contains more than {{maxLayers}} elements the promise MUST be rejected. If {{requestPresent()}} is called outside of an engagement gesture the promise MUST be rejected unless the {{VRDisplay}} was already presenting. This engagement gesture is also sufficient to allow <a href="http://w3c.github.io/pointerlock/">requestPointerLock</a> calls until presentation has ended. The user agent MAY reject the promise for any other reason. If the {{VRDisplay}} is already presenting when {{requestPresent()}} is called the {{VRDisplay}} SHOULD update the {{VRLayerInit}} list being presented. If a call to {{requestPresent()}} is rejected while the {{VRDisplay}} is already presenting the {{VRDisplay}} MUST end presentation.
+Begins presenting the contents of the specified {{VRLayerInit}} array on the {{VRDisplay}} and fulfills the returned promise when presentation has begun. If {{canPresent}} is false the promise MUST be rejected. If the {{VRLayerInit}} array contains more than {{maxLayers}} elements the promise MUST be rejected. If any of the {{VRSource}} properties specified in the {{VRLayerInit}} array contain a context that is not WebGL (i.e. WebGLRenderingContextBase) the promise MUST be rejected. If {{requestPresent()}} is called outside of an engagement gesture the promise MUST be rejected unless the {{VRDisplay}} was already presenting. This engagement gesture is also sufficient to allow <a href="http://w3c.github.io/pointerlock/">requestPointerLock</a> calls until presentation has ended. The user agent MAY reject the promise for any other reason. If the {{VRDisplay}} is already presenting when {{requestPresent()}} is called the {{VRDisplay}} SHOULD update the {{VRLayerInit}} list being presented. If a call to {{requestPresent()}} is rejected while the {{VRDisplay}} is already presenting the {{VRDisplay}} MUST end presentation.
```

Comment:
to improve ambiguity, I'd recommend adding commas for these conditional sentences:

- `is false the` -> `is false, the`
- `elements the promise` -> `elements, the promise`
- `not WebGL (i.e. WebGLRenderingContextBase) the promise` -> `not WebGL (i.e., WebGLRenderingContextBase), the promise`
- `gesture the promise` -> `gesture, the promise`
- `user agent` -> `UA` (?)
- `is called the` -> `is called, the`
- `presenting the` -> `presenting, the`